### ISSUE TITLE
NinjectHttpApplication::Application_End implementation question

### ISSUE THREAD

**Issue by farzadpanahi** [ninject/Ninject.Web.Common#9]
I have a question. At the moment Application_End implementation method for [NinjectHttpApplication.cs](https://github.com/ninject/Ninject.Web.Common/blob/master/src/Ninject.Web.Common/NinjectHttpApplication.cs) looks like:

```
    public void Application_End()
    {
        this.bootstrapper.ShutDown();
        this.OnApplicationStopped();
    }
```

My question is isn't it better to flip the lines to:

```
    public void Application_End()
    {
        this.OnApplicationStopped();
        this.bootstrapper.ShutDown();
    }
```

Here are my reasons: 
1. It is more consistent with Application_Start:
   
   ```
   public void Application_Start()
   {
       lock (this)
       {
           this.bootstrapper.Initialize(this.CreateKernel);
           this.onePerRequestHttpModule.ReleaseScopeAtRequestEnd = this.bootstrapper.Kernel.Settings.Get("ReleaseScopeAtRequestEnd", true);
           this.OnApplicationStarted();
       }
   }
   ```
   
   Here first the bootstrapper is initialized and then OnApplicationStarted is called. So it makes sense to call OnApplicationStopped first and then shutdown the boostrapper. 
   Basically I am trying to say that if the ninject bootstrapper is the first thing that is supposed to be initialized when the application starts, it should be last thing to shutdown when the application ends.
2. Implementer of NinjectHttpApplication (like MvcApplication) can do more on application shutdown. For instance if I need to do something on Application Shutdown that requires ninject dependency resolution, then it only works if bootstrapper is shutdown after OnApplicationStopped, otherwise it would fail because the kernel is already dead. 

RIght now since bootstrapper is shutdown before OnApplicationStopped, basically you cannot do anything that needs ninject resolution.

What do you think?
